dlt/extract/pipe.py

```diff
@@ -697,6 +708,39 @@ def _get_source_item(self) -> ResolvablePipeItem:
         except Exception as ex:
             raise ResourceExtractionError(pipe.name, gen, str(ex), "generator") from ex
 
+
+
+    def _get_source_item_round_robin(self) -> ResolvablePipeItem:
+        try:
+            # get items from last added iterator, this makes the overall Pipe as close to FIFO as possible
+            self._round_robin_index = self._round_robin_index % len(self._sources)
+            gen, step, pipe, meta = self._sources[self._round_robin_index]
+            self._round_robin_index += 1
+            # print(f"got {pipe.name} {pipe._pipe_id}")
+            # register current pipe name during the execution of gen
+            set_current_pipe_name(pipe.name)
+            item = next(gen)
+            # full pipe item may be returned, this is used by ForkPipe step
+            # to redirect execution of an item to another pipe
+            if isinstance(item, ResolvablePipeItem):
+                return item
+            else:
+                # keep the item assigned step and pipe when creating resolvable item
+                if isinstance(item, DataItemWithMeta):
+                    return ResolvablePipeItem(item.data, step, pipe, item.meta)
+                else:
+                    return ResolvablePipeItem(item, step, pipe, meta)
+        except StopIteration:
+            # we need to decrease the index to keep the round robin order
+            self._round_robin_index -= 1
+            # remove empty iterator and try another source
+            self._sources.pop(self._round_robin_index)
+            return self._get_source_item()
```

Comment:
not the round robin again?

Reply:
I changed the code a bit, and it has to stay like this for round robin and current to work in conjunction